File: Python/Program/Laptop.py
from Komputer import Komputer
from Portable import Portable

class Laptop(Komputer, Portable):
    # Constructor
    def __init__(self, nama="", cpu=None, gpu=None, storage=None, battery_capacity=0, weight=0.0, tipe_laptop="", os_laptop=""):
        Komputer.__init__(self, nama, cpu, gpu, storage)
        Portable.__init__(self, battery_capacity, weight)
        self.tipe_laptop = tipe_laptop
        self.os_laptop = os_laptop
    
    # Getters
    def get_tipe_laptop(self):
        return self.tipe_laptop
    
    def get_os_laptop(self):
        return self.os_laptop
    
    # Setters
    def set_tipe_laptop(self, tipe_laptop):
        self.tipe_laptop = tipe_laptop
    
    def set_os_laptop(self, os_laptop):
        self.os_laptop = os_laptop

File: CPP/Program/Laptop.cpp
#pragma once
#include <bits/stdc++.h>
#include "Komputer.cpp"
#include "Portable.cpp"
using namespace std;

class Laptop : public Komputer, public Portable {
    // Attributes
    private:
        string tipeLaptop;
        string osLaptop;
    
    // Constructor
    public:
        // Default constructor
        Laptop() {
            this->tipeLaptop = "";
            this->osLaptop = "";
        }
        
        // Parameterized constructor
        Laptop(string nama, Cpu cpu, GPU gpu, Harddrive storage, int batteryCapacity, float weight, string tipeLaptop, string osLaptop) 
            : Komputer(nama, cpu, gpu, storage), Portable(batteryCapacity, weight) {
            this->tipeLaptop = tipeLaptop;
            this->osLaptop = osLaptop;
        }
        
        // Getters
        string getTipeLaptop() {
            return this->tipeLaptop;
        }
        
        string getOsLaptop() {
            return this->osLaptop;
        }
        
        // Setters
        void setTipeLaptop(string tipeLaptop) {
            this->tipeLaptop = tipeLaptop;
        }
        
        void setOsLaptop(string osLaptop) {
            this->osLaptop = osLaptop;
        }
        
        // Destructor
        ~Laptop() {
        }
};